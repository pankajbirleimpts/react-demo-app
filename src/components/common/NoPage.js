import React from "react";
import { Grid } from "semantic-ui-react";
export default function NoPage() {
  return (
    <Grid centered columns={6} style={{ height: "400px" }}>
      <Grid.Row verticalAlign="middle">
        <Grid.Column>
          <h2 className="ui icon header">
            <i aria-hidden="true" className="settings icon"></i>
            400 Error
            <div className="sub header">The page is not found</div>
          </h2>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  );
}
