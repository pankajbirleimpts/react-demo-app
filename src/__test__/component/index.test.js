import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { reactLocalStorage } from 'reactjs-localstorage';
import Routing, { UnConnectedRouting } from '../../components';
import Login from '../../components/signin/index';
import NoPage from '../../components/common/NoPage';
import { storeFactory } from '../test/TestUtils';



const setup = (props) => {
    const store = storeFactory(props);
    const wrapper = shallow(<Routing store={store} />)
        .dive()
        .dive();
    return wrapper;
};

describe('Tested the Routing', () => {
    const props = {
        user: {
            isAuthenticated: false,
        },
    };
    const store = storeFactory(props);

    test('Snapshot of Routing component', () => {
        const wrapper = setup(props);
        expect(wrapper).toMatchSnapshot();
    });

    test('Invalid path should redirect to 404', () => {
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/random']}>
                    <Routing />
                </MemoryRouter>
            </Provider>,
        );
        expect(wrapper.find(Login)).toHaveLength(0);
        expect(wrapper.find(NoPage)).toHaveLength(1);
    });

    test('Valid path should not redirect to 404', () => {
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/login']}>
                    <Routing />
                </MemoryRouter>
            </Provider>,
        );
        expect(wrapper.find(Login)).toHaveLength(1);
        expect(wrapper.find(NoPage)).toHaveLength(0);
    });

    test('Check the isAuthenticated in props', () => {
        const props = {
            user: {
                isAuthenticated: false,
            },
        };
        const wrapper = setup(props);
        const componentProps = wrapper.instance().props;
        expect(componentProps.user.isAuthenticated).toBe(props.user.isAuthenticated);
    });

    test('Check the updateUserStore action in props', () => {
        const props = {
            user: {
                isAuthenticated: false,
            },
        };
        const wrapper = setup(props);
        const componentProps = wrapper.instance().props;
        expect(componentProps.updateUserStore).toBeInstanceOf(Function);
    });

    test('Check the updateUserStore action should be called if user is authenticated', () => {
        const updateUserStoreMock = jest.fn();
        const props = {
            updateUserStore: updateUserStoreMock,
            user: {
                isAuthenticated: false,
            },
        };
        const wrapper = shallow(<UnConnectedRouting {...props} />);
        // Run life cylce method
        wrapper.instance();
        const loggedUser = reactLocalStorage.get('loggedUser');
        const componentProps = wrapper.instance().props;
        if (loggedUser && componentProps.user.isAuthenticated === false) {
            expect(updateUserStoreMock).toHaveBeenCalled();
        }
    });
});
