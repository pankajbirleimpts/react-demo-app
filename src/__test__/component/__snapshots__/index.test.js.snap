// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tested the Routing Snapshot of Routing component 1`] = `
<Container>
  <Connect(withRouter(Header)) />
  <div
    className="ui segment routing-container"
  >
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(UnConnectedSignin)",
            "type": [Function],
          }
        }
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(UnConnectedSignin)",
            "type": [Function],
          }
        }
        path="/login"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(UnConnectedSignup)",
            "type": [Function],
          }
        }
        path="/signup"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Dashboard)",
            "type": [Function],
          }
        }
        path="/dashboard"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(UnconnectedUserList)",
            "type": [Function],
          }
        }
        path="/users"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(AddUser))",
            "type": [Function],
          }
        }
        path="/add-user"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(AddUser))",
            "type": [Function],
          }
        }
        path="/update-user/:id"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(AddItem))",
            "type": [Function],
          }
        }
        path="/add-item"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(AddItem))",
            "type": [Function],
          }
        }
        path="/update-item/:id"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(ItemList)",
            "type": [Function],
          }
        }
        path="/items"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(DayItem))",
            "type": [Function],
          }
        }
        path="/add-day-item"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(DayItem))",
            "type": [Function],
          }
        }
        path="/update-day-item/:id"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(DayItemList))",
            "type": [Function],
          }
        }
        path="/day-items"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(PurchaseItem))",
            "type": [Function],
          }
        }
        path="/purchase-item"
        permission="ADMIN"
      />
      <Connect(AuthRouteFn)
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(TransactionList))",
            "type": [Function],
          }
        }
        path="/transactions"
      />
      <Route
        component={[Function]}
        path="*"
      />
    </Switch>
  </div>
</Container>
`;
